### ISSUE TITLE
Is it possible to get correct validation errors type back from an action defining schema with `zod-form-data`?

### ISSUE THREAD

**Issue by ervasive** [TheEdoRan/next-safe-action#41]
Currently we get `Partial<Record<"_root", string[]>>` out of an action defining schema with `zod-form-data` instead of the actual fields that were specified, which is cumbersome to use.

Example:

```ts
zfd.formData({
    id: zfd.text(z.string().uuid()),
    name: zfd.text(z.string().min(8).max(100)),
    is_public: zfd.checkbox(),
})
```

Is there a way to get the correct type out of an action with `zfd` schema?

Cheers

**Comment by TheEdoRan** [comment]
Should be fixed in 5.2.2. Please let me know!

**Comment by ervasive** [comment]
Works flawlessly. Thank you for a great library.
